crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -148,6 +150,21 @@ where
             htlc_maximum_msat: None,
         };
 
+        let is_initial_channel_open = !self
+            .channel_manager
+            .list_usable_channels()
+            .iter()
+            .any(|channel| channel.counterparty.node_id == target_node);
+        if is_initial_channel_open {
+            let channel =
+                Channel::new_from_initial_channel_open(target_node, FakeScid::new(intercept_scid));
+            // If we fail to record the channel based on intercept id into the database we don't
+            // fail but print an error
+            if let Err(e) = self.storage.upsert_channel(channel) {
+                tracing::error!("Failed to insert channel based on scid: {e:#}");
+            }
```

Comment:
🔧 I would just fail here. Obviously storing the fake SCID is not super crucial, but I don't see why it should fail anyway.

Reply:
I wonder if the above check is exhaustive enough to say we fail here. I wonder if we can have cases where we don't have a usable channel *just yet*; then the invoice creation would fail (because we cannot insert the same channel twice) event though the user just wants to make a payment. I felt it might be better to not make the invoices creation depend on this insert...

Reply:
I think eventually it would be good to have two different pathes for initial channel open and payments over an existing channel. But I did not want to refactor this as part of this PR...

Reply:
I see where you are coming from.

I think ideally we would only call this function when we know that the invoice will be used to open a JIT channel. Then we wouldn't need to reason about the possibility that this might be for a regular payment.

It's probably okay to leave this as is for now then.